bullet_train-api/app/helpers/api/open_api_helper.rb

```diff
@@ -100,31 +100,54 @@ def automatic_components_for(model, **options)
+        # the bullet train api schema
+        if model.name == "Team" || model.name == "User"
+          create_parameters_output = nil
+        elsif model.name == "Invitation"
+          update_parameters_output = nil
         end
 
-        parameters_output = JSON.parse(schema_json)
-        parameters_output["required"].select! { |key| strong_parameter_keys.include?(key.to_sym) }
-        parameters_output["properties"].select! { |key| strong_parameter_keys.include?(key.to_sym) }
-        parameters_output["example"]&.select! { |key, value| strong_parameter_keys.include?(key.to_sym) && value.present? }
-
-        # Allow customization of Parameters
-        customize_component!(parameters_output, options[:parameters]) if options[:parameters]
-
-        (
-          indent(attributes_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}Attributes:"), 3) +
-            indent("    " + parameters_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}Parameters:"), 3)
-        ).html_safe
+        output = indent(attributes_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}Attributes:"), 3)
+        output += indent("    " + create_parameters_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}ParametersCreate:"), 3) if create_parameters_output
+        output += indent("    " + update_parameters_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}ParametersUpdate:"), 3) if update_parameters_output
+        output.html_safe
       else
 
         indent(attributes_output.to_yaml.gsub("---", "#{model.name.gsub("::", "")}Attributes:"), 3)
           .html_safe
       end
     end
 
+    def process_strong_parameters(strong_params_module, schema_json, method_type, **options)
+      strong_parameter_keys = strong_params_module.report(method_type)
+      strong_parameter_keys += strong_parameter_keys.pop.keys if strong_parameter_keys.last.is_a?(Hash)
+
+      parameters_output = JSON.parse(schema_json)
+      parameters_output["required"].select! { |key| strong_parameter_keys.include?(key.to_sym) }
+      parameters_output["properties"].select! { |key| strong_parameter_keys.include?(key.to_sym) }
+      parameters_output["example"]&.select! { |key, value| strong_parameter_keys.include?(key.to_sym) }
+
+      # Allow customization of Parameters
+      parameters_custom = options[:parameters][method_type] if options[:parameters].is_a?(Hash) && options[:parameters].key?(method_type)
+      parameters_custom ||= options[:parameters]
+      customize_component!(parameters_output, parameters_custom, strong_params_module.model_name.underscore, method_type) if parameters_custom
+
+      # We need to wrap the example parameters with the model name as expected by the API controllers
+      if parameters_output["example"]
+        parameters_output["example"] = {strong_params_module.model_name.underscore.tr("/", "_") => parameters_output["example"]}
```

Comment:
the model_name is coming from `def model_name` method added to the `strong_parameters_reporter, but I will change it to just expose model then it will have consistent naming and can use `strong_params_module.model.name.underscore` and `strong_params_module.model.param_key`

Reply:
Ah, I didn't notice it was declared, I thought it is this one: https://api.rubyonrails.org/classes/ActiveModel/Naming.html